surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+- Desktop computer or mobile phone libraries or services
+- Distributed systems
+- Embedded devices (with operating systems)
+- Embedded devices (bare metal)
+- HPC (High-performance [Super]Computing)
+- IoT (Internet of Things)
+- Machine learning or AI
+- Mobile phone application frontend
+- Programming languages and related tools (including compilers, IDEs, standard libraries, etc.)
+- Robotics
+- Scientific and/or numerical computing
+- Server-side or "backend" application
+- Simulation
+- Web application frontend
+- WebAssembly
+- Other (open response)
+
+> **justification**
+>
+> We want to known roughly what technology stacks are being most often used.
+>
+> This can be ambiguous and hard to answer. For example, if you're building an operating
+> system for a mobile phone, is that embedded, mobile, or something else?
+> We want to understand the "shape" of Rust usage, and this question tries to get at that
+> by allowing the respondent to select multiple answers.
+
+### Approximately how many total developers does your company employ?
+
+**Note**: don't worry about being exact here! Go with you gut.
+
+Type: select one (optional)
+
+- Under 10
+- 10-99
+- 100-1,000
+- 1,000-10,000
+- Over 10,000
+
+> This question is not that interesting on its own, but it can be used as a sort of cohort for understanding how answers
+> change depending on the size of the development effort at a company.
+>
+> Previously this question used "employees" instead of "developers". It is more appropriate for us to ask about the amount
+> of developers at a company vs. the amount of people employed in total.
+
+### Is your company planning on hiring Rust developers in the next year?
+
+Type: select one (optional)
+
+- Yes
+- No
+- I don't know
+
+> This question assess hiring sentiment. Although there is intrinsic uncertainty, it is easy to answer and forward looking.
+> It will also be interesting to see what the demand for Rust skills from companies is over time.
+
+## Your opinions about Rust
+
+### Which of the following statements about Rust do you feel are true?
+
+Type: select all that apply (optional)
+
+Statements:
+
+- Rust provides a real benefit over other programming languages
+- Rust is significantly more complicated to program in than other programming languages
+- Rust requires significantly more effort to learn than other programming languages
+- Rust code tends to contain significantly fewer bugs than equivalent code written in another programming language would have
+- Rust is risky to use in production
+- Rust makes me more productive
+- Rust is fun to use
+
+> **justification**
+>
+> There are several "truisms" about Rust that we'd like to get perspective on how true these are for users of Rust.
+>
+> Note that answers here can be subject to survivorship bias and so extra care should be taken with interpreting results.
+
+
+### What are your biggest worries for the future of Rust?
+
+Type: select all that apply (optional)
```

Comment:
Fears around governance (both the project and the foundation) are the biggest concerns I've heard, and there's no good option to record them here.

Reply:
> "Project governance does not scale to match the size/requirements of the community"

How would you formulate an additional question to this one reflecting those concerns?

Reply:
"Project governance is incompetent" or "Project governance is not aligned with the goals of the Rust community" are two flavors of concerns that I think are much more accurate to what I've heard expressed.

Reply:
I agree that more options regarding this (very publicly visible concern) should be added. I'm not sure what's the best way to fomulate it though. The "Project governance is not aligned with the goals of the Rust community" sounds reasonable to me.

What about the distinction between the Rust Project/Rust foundation/Leadership chat/Leadership council? Do we want to somehow accent it here?

Reply:
I think "Rust leadership" is probably best. I don't really think that the average community member has a clear differentiation here.

Reply:
> I don't really think that the average community member has a clear differentiation here.

that's a very good point and _exactly_ for this reason IMO worth underlining here the difference: Rust Foundation != [Rust Governance](https://www.rust-lang.org/governance) (i.e. us, the project). I'd prefer ensuring that people are referring to the correct entity when expressing concerns.

So my proposal to implement Alices' suggestion would be:
> * Rust Foundation is not doing a good job at supporting the Rust project (in any of its aspects: financial, infrastructure, legal, too much corporate, etc.)

Concerns towards the Rust Governance (i.e. us :sweat_smile: ) are perhaps covered by the existing option:
> *Project governance does not scale to match the size/requirements of the community

(feel free to comment and reword, ofc)

Reply:
I agree with the Rust Foundation answer, but I'm not sure if it's "enough" to keep the "Project governance does not scale..." as a way to voice concerns regarding Rust governance. For example, see [this complaint](https://www.reddit.com/r/rust/comments/15klpl2/comment/jv6lxhc/?utm_source=share&utm_medium=web2x&context=3) made in relation with the last year's survey.

There have been three major public Rust uproars this year (AFAIK, it's hard to keep track :sweat_smile:).
- Trademark -> belongs mainly under Rust Foundation concerns, at least from outsider's view, IMO
- RustConf fiasco -> not sure if can be described as "governance does not scale", perhaps yes (?)
- Serde precompiled binaries -> probably neither a governance nor a foundation concern, but maybe could be described by some other answer (ownership/bus factor of foundational Rust crates?).

Reply:
Regarding the foundation answer, I'd suggest rewording to
```
- Rust Foundation not supporting the Rust project properly (e.g. in financial, infrastructure, legal aspects)
```
to keep it shorter and more inline with the other answers (using a present tense there was a bit weird to me).

Reply:
> I agree with the Rust Foundation answer, but I'm not sure if it's "enough" to keep the "Project governance does not scale..." as a way to voice concerns regarding Rust governance. 

Hmm yes. hard for a single option to encompass all aspects where the governance can fail part of the community. Let's see, how does that sound:

> * Project governance not properly and/or timely handling issues affecting the community

(or something along this line...)